IntegrationTests/conftest.py

```diff
@@ -0,0 +1,105 @@
+"""
+Defines variables that are required for a report to be generated.
+"""
+
+import os
+import yaml
+
+# Report Generate Logic
+tests_failed_unexpectedly = []
+tests_passed_unexpectedly = []
+tests_passed = 0  # pylint: disable=invalid-name
+tests_failed = 0  # pylint: disable=invalid-name
+total_tests = 0  # pylint: disable=invalid-name
+tests_skipped = 0  # pylint: disable=invalid-name
+
+# Lists containing references to each exception test
+known_failures = []
+failing_as_unspecified = []
+unsupported = []
+do_not_run = []
+
+# Environment variables
+DEBUG = False
+CLASSPATH = ""
+EXECUTABLE = ""
+GENERATE_REPORT_LOCATION = ""
+TEST_EXCEPTIONS = ""
+GENERATE_REPORT = False
+
+
+# Log data
+def pytest_runtest_logreport(report):
+    """
+    Increment the counters for what action was performed
+    """
+    # Global required here to access counters
+    # Not ideal but without the counters wouldn't work
+    global total_tests, tests_passed, tests_failed, tests_skipped  # pylint: disable=global-statement
+    if report.when == "call":  # only count test function execution, not setup/teardown
+        total_tests += 1
+        if report.passed:
+            tests_passed += 1
+        elif report.failed:
+            tests_failed += 1
+        elif report.skipped:
+            tests_skipped += 1
+
+
+# Run after all tests completed, Generate a report of failing and passing tests
+def pytest_sessionfinish(exitstatus):
+    """
+    Generate report based on test run
+    """
+    if GENERATE_REPORT:
+        print("Generating report for the test run")
+        os.makedirs(GENERATE_REPORT_LOCATION, exist_ok=True)
```

Comment:
Why does this need to be a directory? Could `GENERATE_REPORT` not just take the file path of the file that is to be created?